dotCMS/src/test/java/com/dotcms/content/elasticsearch/ESQueryCacheTest.java

```diff
@@ -0,0 +1,272 @@
+        assertTrue(cache.get(request).isPresent());
+        
+        assertEquals(hits, cache.get(request).get());
+        
+        SearchRequest request2 = getSearchRequest();
+        assertTrue(!cache.get(request2).isPresent());
+
+    }
+    
+    
+    @Test
+    public void test_get_hits_by_request() {
+
+        final String testQuery =RandomStringUtils.randomAscii(200);
+        SearchHits hits= hits();
+        
+        SearchRequest req1 = getSearchRquest(testQuery, true);
+        SearchRequest req2 = getSearchRquest(testQuery, true);
+        cache.put(req1, hits);
+        
+
+        assert(cache.get(req1).isPresent());
+        
+        assert(cache.get(req2).isPresent());
+        
+        assertEquals(cache.get(req1).get(), cache.get(req2).get());
+
+    }
+    
+    
+    @Test
+    public void test_clear_cache() {
+
+        final String testQuery =RandomStringUtils.randomAscii(200);
+        SearchHits hits= hits();
+        
+        SearchRequest req1 = getSearchRquest(testQuery, true);
+        SearchRequest req2 = getSearchRquest(testQuery, true);
+        cache.put(req1, hits);
+        
+
+        
+        assert(cache.get(req1).isPresent());
+        assert(cache.get(req2).isPresent());
+        
+        cache.clearCache();
+        
+        assert(!cache.get(req1).isPresent());
+        assert(!cache.get(req2).isPresent());
+
+
+    }
+    
+    
+    
+    
+    SearchHits hits() {
+        Random rand = new Random(); 
+        List<SearchHit> hitList = new ArrayList<>();
+        for(int i=0;i< rand.nextInt(10);i++) {
+            Text text = new Text(RandomStringUtils.randomAscii(36));
+            SearchHit hit = new SearchHit(i,RandomStringUtils.randomAscii(36),text,new HashMap<>());
+            hitList.add(hit);
+        }
+        
+        return new SearchHits(hitList.toArray(new SearchHit[0]), new TotalHits(2, Relation.EQUAL_TO), .7f);
+        
+        
+    }
+    
+    
+    
+    private SearchRequest getSearchRequest() {
+
+        return getSearchRquest(RandomStringUtils.randomAscii(200), true) ;
+        
+    }
+    
+    @NotNull
+    private SearchRequest getSearchRquest(String query, boolean live) {
+        SearchRequest searchRequest = new SearchRequest();
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.query(QueryBuilders.queryStringQuery(query));
+        searchSourceBuilder.size(0);
+        searchSourceBuilder.timeout(TimeValue.timeValueMillis(INDEX_OPERATIONS_TIMEOUT_IN_MS));
+        searchRequest.source(searchSourceBuilder);
+        searchRequest.indices((live ? "live" : "working"));
+        return searchRequest;
+    }
+
+    static DotCacheAdministrator admin = new DotCacheAdministrator() {
+
+        private Map<String, SearchHits> hitMap = new HashMap<>();
+
+        @Override
+        public void shutdown() {
+
+
+        }
+
+        @Override
+        public void setTransport(CacheTransport transport) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'transport' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5288908)